### PR TITLE
BFB-275 : Update to Unity 2021

### PR DESCRIPTION
**Pull request by unoctium1**
**JIRA Ticket**
https://perigongames.atlassian.net/browse/BFB-275

**Description**
Update Unity to 2021.3.0f1
Also includes updates to mixpanel and most other packages in use

Note: Due to a urp compatibility issue, there's a regression and dash lines are gone pending an implementation change

### CODE REVIEW

File: ProjectSettings/RiderScriptEditorPersistedState.asset [hunk 1]
```diff
@@ -0,0 +1,15 @@
+%YAML 1.1
```
Comment (m4rtint) [inline ProjectSettings/RiderScriptEditorPersistedState.asset:1]: I don't think this is needed is it?

**Comment by m4rtint** [comment]
Are all the materials changes part of the update?

File: Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs [hunk 2]
```diff
@@ -48,7 +48,7 @@ public ObjectPooler(string poolName, Func<T> buildObject, Action<T> actionOnGet
 		~ObjectPooler()
         {
 #if UNITY_2021_1_OR_NEWER
-			objectPool.Dispose();
+			objectPool.Clear();
 #else
 			SceneManager.UnloadSceneAsync(poolScene); //destroys all objects in the pool
```
Comment (m4rtint) [inline Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs:53]: Shouldn't the rest of the code be removed?

**Comment by unoctium1** [comment]
> Are all the materials changes part of the update?

Yeah, cause URP is updated as well. They aren't meaningfully changed, just reserialized with the new version

File: Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs (hunk 2 above)
Comment (unoctium1) [reply Assets/_BForBoss/_Utility/Scripts/ObjectPooler.cs:53]: Sure I'll do that

File: ProjectSettings/RiderScriptEditorPersistedState.asset (hunk 1 above)
Comment (unoctium1) [reply ProjectSettings/RiderScriptEditorPersistedState.asset:1]: Like ignore it? I can try that, I just as a rule thought new project settings files should be added

**Review by mikeslgoh — APPROVED** [review]

**Pull request merged**